route-rs-runtime/src/link/primitive/process_link.rs

```diff
@@ -2,6 +2,10 @@ use crate::link::{Link, LinkBuilder, PacketStream, ProcessLinkBuilder};
 use crate::processor::Processor;
 use futures::{Async, Poll, Stream};
 
+/// `ProcessLink` is a simple links used to process packets through the user-defined processor
+/// It does not have the ability to store packets internally, and is pull-based. So it only does
+/// work when it is called, and must either immediatly return recieved packets, or drop them.  This
+/// link is used for transformation logic.
```

Comment:
"It cannot buffer packets, so it only does work when it is called. It must immediately drop or return a transformed packet."

We can remove the last sentence.